nautobot_ssot/tests/ipfabric/test_ipfabric_adapter.py

```diff
@@ -83,3 +83,28 @@ def test_data_loading(self):
             self.assertTrue(hasattr(interface, "ip_address"))
             self.assertTrue(hasattr(interface, "subnet_mask"))
             self.assertTrue(hasattr(interface, "type"))
+
+    @patch("nautobot_ssot.integrations.ipfabric.diffsync.adapter_ipfabric.IP_FABRIC_ELONGATE_INTERFACE_NAME", True)
```

Comment:
Not requiring to swap out but would be good to patch the setting instead of the constant. Should be the same net effect. TBH how we do constants for plugin config kinda negates the lazy loading part of Django settings BUT that pattern is everywhere and we don't document or expect users to relying on lazy loading

Reply:
ya, I'm not a fan either, but following convention. As for patching the setting, as far as I know, I would have to patch all of `nautobot_ssot` settings, which seems a bit excessive; is there another solution?

Reply:
Yes it's a patch of all plugin config but all tests should run against the default config so if you pass in just the one toggle that's all that changed.

Reply:
Currently tests are being impacted by the settings in PLUGINS_CONFIG so it might be necessary to patch the specific setting. We are planning on addressing that issue in the near future.

Reply:
We should fix the tests to be in a state that do not have to have the settings changed.

Reply:
I agree and it's on the todo list.

Reply:
I disagree on this one, this is testing a setting change so it should be tested from the PLUGIN_CONFIG side. Intercepting the constant definition could mask an improperly defined plugin config in the NautobotAppConfig.

I am okay resolving this item, whatever we do in the future should test making the setting change in nautobot config and not patching elsewhere. Django lazy loading + default values for plugin config + patching the constant that pulls out a value from plugin config could be a false sense of coverage. I would vote this is a feature request into the nautobot testing framework.

Reply:
actually, this is testing whatever the value of the constant is since that is what is imported/used. There should be a separate set of tests for constants.py that tests the constants are set correctly